libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraComponent.kt

```diff
@@ -188,47 +189,51 @@ internal class CameraComponent constructor(
     }
 
     private fun onRouteUpdates(mapboxNavigation: MapboxNavigation) {
-        coroutineScope.launch {
-            val routesFlow = mapboxNavigation.flowRoutesUpdated()
+        val routesFlow = combine(
+            store.select { it.previewRoutes },
+            mapboxNavigation.flowRoutesUpdated()
                 .map { it.navigationRoutes }
                 .stateIn(
                     coroutineScope,
                     SharingStarted.WhileSubscribed(),
-                    mapboxNavigation.getNavigationRoutes()
-                )
-
-            val routePreview = store.select { it.previewRoutes }
-            val navigation = store.select { it.navigation }
-            combine(
-                routePreview,
-                routesFlow,
-                navigation
-            ) { previewRoutes, navigationRoutes, navigationState ->
-                val routes = when {
-                    navigationRoutes.isNotEmpty() -> navigationRoutes
-                    previewRoutes is RoutePreviewState.Ready -> previewRoutes.routes
-                    else -> emptyList()
+                    mapboxNavigation.getNavigationRoutes(),
+                ),
+        ) { previewRoutes, navigationRoutes ->
+            when {
+                navigationRoutes.isNotEmpty() -> navigationRoutes
+                previewRoutes is RoutePreviewState.Ready -> previewRoutes.routes
+                else -> emptyList()
+            }
+        }
+        routesFlow.observe { routes ->
+            if (routes.isEmpty()) {
+                viewportDataSource.clearRouteData()
+                viewportDataSource.evaluate()
+            } else {
+                viewportDataSource.onRouteChanged(routes.first())
+                viewportDataSource.evaluate()
+            }
+        }
+        val navigationStateFlow = store.select { it.navigation }
+        combine(navigationStateFlow, routesFlow) { navigationState, routes ->
+            navigationState == NavigationState.FreeDrive && routes.isEmpty()
+        }.filter { it }.observe { store.dispatch(SetCameraMode(TargetCameraMode.Overview)) }
```

Comment:
This store.dispatch would be better moved to the other flowable.

There are 2 streams dispatching camera modes, it's easier to understand when there is 1 stream that will dispatch the events